src/main/java/org/opensearch/ad/rest/handler/AbstractAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,821 @@
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryBuilders;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.search.aggregations.AggregatorFactories;
+import org.opensearch.search.builder.SearchSourceBuilder;
+import org.opensearch.transport.TransportService;
+
+public abstract class AbstractAnomalyDetectorActionHandler<T extends ActionResponse> {
+    public static final String EXCEEDED_MAX_MULTI_ENTITY_DETECTORS_PREFIX_MSG = "Can't create multi-entity anomaly detectors more than ";
+    public static final String EXCEEDED_MAX_SINGLE_ENTITY_DETECTORS_PREFIX_MSG = "Can't create single-entity anomaly detectors more than ";
+    public static final String NO_DOCS_IN_USER_INDEX_MSG = "Can't create anomaly detector as no document found in indices: ";
+    public static final String ONLY_ONE_CATEGORICAL_FIELD_ERR_MSG = "We can have only one categorical field.";
+    public static final String CATEGORICAL_FIELD_TYPE_ERR_MSG = "A categorical field must be of type keyword or ip.";
+    public static final String NOT_FOUND_ERR_MSG = "Cannot found the categorical field %s";
+    // Modifying message for FEATURE below may break the parseADValidationException method of ValidateAnomalyDetectorTransportAction
+    public static final String FEATURE_INVALID_MSG_PREFIX = "Feature has invalid query";
+    public static final String FEATURE_WITH_EMPTY_DATA_MSG = FEATURE_INVALID_MSG_PREFIX + " returning empty aggregated data: ";
+    public static final String FEATURE_WITH_INVALID_QUERY_MSG = FEATURE_INVALID_MSG_PREFIX + " causing runtime exception: ";
+    public static final String UNKNOWN_SEARCH_QUERY_EXCEPTION_MSG = "Unknown exception caught while searching query for feature: ";
+
+    protected final AnomalyDetectionIndices anomalyDetectionIndices;
+    protected final String detectorId;
+    protected final Long seqNo;
+    protected final Long primaryTerm;
+    protected final WriteRequest.RefreshPolicy refreshPolicy;
+    protected final AnomalyDetector anomalyDetector;
+    protected final ClusterService clusterService;
+
+    protected final Logger logger = LogManager.getLogger(AbstractAnomalyDetectorActionHandler.class);
+    protected final TimeValue requestTimeout;
+    protected final Integer maxSingleEntityAnomalyDetectors;
+    protected final Integer maxMultiEntityAnomalyDetectors;
+    protected final Integer maxAnomalyFeatures;
+    protected final AnomalyDetectorActionHandler handler = new AnomalyDetectorActionHandler();
+    protected final RestRequest.Method method;
+    protected final Client client;
+    protected final TransportService transportService;
+    protected final NamedXContentRegistry xContentRegistry;
+    protected final ActionListener<T> listener;
+    protected final User user;
+    protected final ADTaskManager adTaskManager;
+    protected final SearchFeatureDao searchFeatureDao;
+    protected final boolean isDryRun;
```

Comment:
Because this instance variable is set upon creation, can we just utilize the value of `this.isDryRun` where it is actually needed, rather than passing it down through several layers of method calls?

Seems the majority of the methods are simply passing the var down to other methods, and it's only really needed in a few places:
- Actually indexing vs. not indexing on [lines 557-558](https://github.com/opensearch-project/anomaly-detection/blob/1659807b945810edcdaae644912b6d7b61dc0b6e/src/main/java/org/opensearch/ad/rest/handler/AbstractAnomalyDetectorActionHandler.java#L557-L558))
- Parsing the responses after creating ([line 359](https://github.com/opensearch-project/anomaly-detection/blob/1659807b945810edcdaae644912b6d7b61dc0b6e/src/main/java/org/opensearch/ad/rest/handler/AbstractAnomalyDetectorActionHandler.java#L359), [line 369](https://github.com/opensearch-project/anomaly-detection/blob/1659807b945810edcdaae644912b6d7b61dc0b6e/src/main/java/org/opensearch/ad/rest/handler/AbstractAnomalyDetectorActionHandler.java#L369))

Reply:
@amitgalitz minor ping on this ^ any thoughts? No blocker for me but curious on your thoughts. If agreed this is a good idea, can you open an issue to track for next release?

Reply:
Hi I looked into this and since then I have added a few more check for indexing variable but utilizing `this.isDryRun` could reduce some unnecessary passing down of the var. I meant to come back to this but totally forgot while fixing other bugs. I will review again to make sure calling`this.isDryRun` is cleaner with new added checks and create an issue.